packages/faust-nx/src/components/WordPressTemplate.tsx

```diff
@@ -2,40 +2,48 @@ import React, { PropsWithChildren } from 'react';
 import { DocumentNode, useQuery } from '@apollo/client';
 import { getTemplate } from '../getTemplate.js';
 import { SeedNode } from '../queries/seedQuery.js';
+import { WordPressTemplate } from '../getWordPressProps.js';
 import { getConfig } from '../config/index.js';
+import { usePreviewQuery } from '../hooks/usePreviewQuery.js';
 
 export type WordPressTemplateProps = PropsWithChildren<{
   __SEED_NODE__: SeedNode;
+  context?: { asPreview?: boolean };
 }>;
 
 export function WordPressTemplate(props: WordPressTemplateProps) {
+  const { __SEED_NODE__: seedNode, context } = props;
   const { templates } = getConfig();
 
   if (!templates) {
     throw new Error('Templates are required. Please add them to your config.');
   }
 
-  const { __SEED_NODE__: seedNode } = props;
   const template = getTemplate(seedNode, templates);
 
   /**
    * This code block exists above the !template conditional
    * as React Hooks can not be behind conditionals
    */
-  const res = useQuery(template?.query as DocumentNode, {
-    variables: template?.variables ? template?.variables(seedNode) : undefined,
-    ssr: true,
-    skip: !template?.query,
-  });
+  let response;
+
+  if (context?.asPreview) {
```

Comment:
We may need to take a closer look at this when the time comes to merge this into `canary` but should be fine for now.

Not sure if this violates the rules of React Hooks since `usePreviewQuery` and `useQuery` are both behind conditionals.

Reply:
I initially thought it would but React didn't throw any errors. @theodesp any thoughts here?

Reply:
You will have to guarantee that only one hook gets called on each render and not the other.
You can pass on a skip parameter in `usePreviewQuery` and check within the `useEffect` if its true then return an empty response

```javascript
if (ctx?.asPreview) {
    response = usePreviewQuery(seedNode, template, false); // last boolean means do not skip
  } else {
    response = useQuery(template?.query as DocumentNode, {
      variables: template?.variables ? template?.variables(seedNode, ctx) : undefined,
      ssr: !ctx?.asPreview,
      skip: !template?.query && ctx?.asPreview,
    });
  }
```

_usePreviewQuery.ts_
```javascript
...
useEffect(() => {
 if ( skip ) {
 return previewResponse;
 }
}
...
```